File: src/app/MoneyApp.java
package app;

import model.Money;
import service.MoneyService;
import view.BarView;
import view.MenuView;

import java.util.Scanner;

public class MoneyApp {
    public static void run (Money money, Scanner sc) {
        boolean conditionRunning = true;
        while (conditionRunning) {
            BarView.bar("Money");
            MenuView.printMenu("Status", "Deposit", "Withdraw", "Exit");
            while (!sc.hasNextInt()) {
                System.out.println("Please enter a valid number");
                sc.next();
            }
            int choice = sc.nextInt();
            switch (choice) {
                case 1:
                    BarView.bar("Status");
                    MoneyService.printMoney(money);
                    break;
                case 2:
                    BarView.bar("Deposit");
                    MoneyService.deposit(money, sc);
                    break;
                case 3:
                    BarView.bar("Withdrawal");
                    MoneyService.withdrawal(money, sc);
                    break;
                case 4:
                    conditionRunning = false;
                    System.out.println("Bye");
                    break;
                default:
                    System.out.println("RETRY");
            }
        }
    }
}

File: src/app/money.app.ts
import { Money } from "../model/money.model";
import { deposit, printMoney, withdrawal } from "../service/money.service";
import { myPrompt } from "../util/my.prompt.util";
import { bar } from "../view/bar.view";
import { printMenu } from "../view/menu.view";

export const runMoneyApp = async (money: Money) => {
  let conditionRunning: boolean = true;
  while (conditionRunning) {
    bar("Money")
    printMenu("Status", "Deposit", "Withdraw", "Exit");
    let input = await myPrompt("Choice: ");
    let choice: number = Number(input);
    while (!choice) {
      choice = Number(await myPrompt("Please enter a valid number\nChoice: "));
    }
    switch (choice) {
      case 1:
        bar("Status");
        printMoney(money);
        break;
      case 2:
        bar("Deposit");
        await deposit(money)
          .then((res) => console.log(res))
          .catch((err) => console.log(err.message));
        break;
      case 3:
        bar("Withdraw");
        await withdrawal(money)
          .then((res) => console.log(res))
          .catch((err) => console.log(err.message));
        break;
      case 4:
        conditionRunning = false;
        console.log("Bye");
        break;
      default:
        console.log("RETRY");
    }
  }
};
